pkg/cableengine/ipsec/strongswan.go

```diff
@@ -291,15 +315,64 @@ func (i *strongSwan) loadSharedKey(endpoint types.SubmarinerEndpoint, client *go
 	return nil
 }
 
-func runCharon(debug bool, logFile string) error {
+// charonConfTemplate defines the configuration for strongswan IKE keying daemon,
+// * port and port_nat_t define the IKE and IKE NATT UDP ports
+// * make_before_break ensures dataplane connectivity while re-authenticating endpoints (check this)
+// TODO : check what * ignore_acquire_ts means
+const charonConfTemplate = `
+	charon {
+		port = {{.ipSecIKEPort}}
+		port_nat_t = {{.ipSecNATTPort}}
+		make_before_break = yes
+		ignore_acquire_ts = yes
+	}
+`
+
+func (i *strongSwan) writeCharonConfig(path string) {
+	err := os.Remove(path)
+	if err != nil {
+		klog.Infof("Error deleting %s", path)
+	}
+
+	f, err := os.Create(path)
+
+	if err != nil {
+		klog.Fatalf("Error creating %s: %s", path, err)
```

Comment:
I think we should propagate an error for consistency. Same with the other `Fatalf` calls below.